packages/packagers/js/src/index.js

```diff
@@ -32,34 +31,40 @@ const CONFIG_SCHEMA: SchemaEntity = {
 
 export default (new Packager({
   async loadConfig({config, options}): Promise<JSPackagerConfig> {
-    // Generate a name for the global parcelRequire function that is unique to this project.
-    // This allows multiple parcel builds to coexist on the same page.
-    let pkg = await config.getConfigFrom(
-      path.join(options.projectRoot, 'index'),
-      ['package.json'],
-    );
-
     let packageKey = '@parcel/packager-js';
+    let conf = await config.getConfigFrom(options.projectRoot + '/index', [], {
+      packageKey,
+    });
 
-    if (pkg?.contents[packageKey]) {
+    if (conf?.contents) {
       validateSchema.diagnostic(
         CONFIG_SCHEMA,
         {
-          data: pkg?.contents[packageKey],
-          source: await options.inputFS.readFile(pkg.filePath, 'utf8'),
-          filePath: pkg.filePath,
+          data: conf?.contents,
+          source: await options.inputFS.readFile(conf.filePath, 'utf8'),
```

Comment:
I think we talked about this during pairing, and it's definitely different for the transformer that does this, but this will only read the `package.json` here once when the packager is initialised, not per package?